flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -254,55 +272,89 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
   }
 
   Future<void> _finishTimeToDisplayTracking() async {
-    _timeToDisplayTracker?.clear();
+    try {
+      final transaction = _transaction;
+      _transaction = null;
+      if (transaction == null || transaction.finished) {
+        return;
+      }
 
-    final transaction = _transaction;
-    _transaction = null;
-    if (transaction == null || transaction.finished) {
-      return;
+      // Cancel unfinished child spans, e.g this might happen if the user navigates
+      // away from the current route before TTFD or TTID is finished.
+      for (final child in (transaction as SentryTracer).children) {
+        if (!child.finished) {
+          await child.finish(status: SpanStatus.cancelled());
+        }
+      }
+
+      transaction.status ??= SpanStatus.ok();
+      await transaction.finish();
+    } catch (exception, stacktrace) {
+      _hub.options.logger(
+        SentryLevel.error,
+        'Error while finishing time to display tracking',
+        exception: exception,
+        stackTrace: stacktrace,
+      );
+    } finally {
+      _clear();
     }
-    transaction.status ??= SpanStatus.ok();
-    await transaction.finish();
   }
 
-  Future<void> _startTimeToDisplayTracking(Route<dynamic>? route) async {
-    if (!_enableAutoTransactions) {
-      return;
-    }
+  Future<void> _finishThenStartTimeToDisplayTracking(
+      Route<dynamic>? route) async {
+    try {
+      // We can await inside this function so we can make sure the previous
+      // tracking is finished before starting a new one.
+      await _finishTimeToDisplayTracking();
```

Comment:
if we remove the previous transaction from the scope, we don't need to await for finishing it